src/ostree/ot-admin-builtin-pin.c

```diff
@@ -54,32 +119,23 @@ ot_admin_builtin_pin (int argc, char **argv, OstreeCommandInvocation *invocation
       char *endptr = NULL;
 
       errno = 0;
-      const guint64 deploy_index = g_ascii_strtoull (deploy_index_str, &endptr, 10);
+      gint64 deploy_index = g_ascii_strtoull (deploy_index_str, &endptr, 10);
       if (*endptr != '\0')
-        return glnx_throw (error, "Invalid index: %s", deploy_index_str);
-      if (errno == ERANGE)
+        {
+          if (!g_strcmp0 (deploy_index_str, "booted"))
```

Comment:
Hmm I'd say we should check the string values first, and then only fall back to trying to parse as an integer.

Reply:
Sure...